mono/metadata/threadpool.c

```diff
@@ -352,10 +354,13 @@ worker_callback (void)
 
 		domains_unlock ();
 
-		MonoString *thread_name = mono_string_new_checked (mono_get_root_domain (), "Thread Pool Worker", error);
-		mono_error_assert_ok (error);
-		mono_thread_set_name_internal (thread, thread_name, MonoSetThreadNameFlag_Reset, error);
-		mono_error_assert_ok (error);
+		if (set_thread_name) {
+			MonoString *thread_name = mono_string_new_checked (mono_get_root_domain (), "Thread Pool Worker", error);
```

Comment:
Could we also avoid allocation this? Maybe introduce a new MonoSetThreadNameFlag_Init flag which only sets thread flags and uses static UTF data for the native name?

Reply:
Yes. I have that in another PR already.

Reply:
I think @jaykrell wanted `MonoSetThreadNameFlag_Constant` to do precisely that.  not sure if one of his open PRs already does it.

Reply:
https://github.com/mono/mono/pull/15945
and then the rest of this:
https://github.com/mono/mono/pull/15859 which was requested to be split up and is almost done.